src/simsopt/field/magneticfieldclasses.py

```diff
@@ -502,6 +503,179 @@ def from_dict(cls, d, serial_objs_dict, recon_objs):
+                dipole_grid_z[index:index + n] = oz * stell
+
+                # get new dipole vectors by flipping the x component, then rotating by phi0
+                m_vec[index:index + n, 0] = mmx * np.cos(phi0) * stell - mmy * np.sin(phi0)
+                m_vec[index:index + n, 1] = mmx * np.sin(phi0) * stell + mmy * np.cos(phi0)
+                m_vec[index:index + n, 2] = mmz
+
+                m_maxima[index:index + n] = pm_opt.m_maxima
+                index += n
+
+        self.dipole_grid = np.array([dipole_grid_x, dipole_grid_y, dipole_grid_z]).T
+        self.m_vec = m_vec
+        self.m_maxima = m_maxima
+
+    def _toVTK(self, vtkname):
+        """
+            Write dipole data into a VTK file (stolen from Caoxiang's CoilPy code).
+
+        Args:
+            vtkname (str): VTK filename, will be appended with .vts or .vtu.
+        """
+
+        # get the coordinates
+        ox = np.ascontiguousarray(self.dipole_grid[:, 0])
+        oy = np.ascontiguousarray(self.dipole_grid[:, 1])
+        oz = np.ascontiguousarray(self.dipole_grid[:, 2])
+        ophi = np.arctan2(oy, ox)
+        otheta = np.arctan2(oz, np.sqrt(ox ** 2 + oy ** 2) - self.pm_opt.R0)
+
+        # define the m vectors and the normalized m vectors
+        # in Cartesian, cylindrical, and simple toroidal coordinates.
+        mx = np.ascontiguousarray(self.m_vec[:, 0])
+        my = np.ascontiguousarray(self.m_vec[:, 1])
+        mz = np.ascontiguousarray(self.m_vec[:, 2])
+        mmag = np.sqrt(mx ** 2 + my ** 2 + mz ** 2)
+        mx_normalized = np.ascontiguousarray(mx / self.m_maxima)
+        my_normalized = np.ascontiguousarray(my / self.m_maxima)
+        mz_normalized = np.ascontiguousarray(mz / self.m_maxima)
+        mr = np.ascontiguousarray(mx * np.cos(ophi) + my * np.sin(ophi))
+        mrminor = np.ascontiguousarray(mx * np.cos(ophi) * np.cos(otheta) + my * np.sin(ophi) * np.cos(otheta) + np.sin(otheta) * mz)
+        mphi = np.ascontiguousarray(-mx * np.sin(ophi) + my * np.cos(ophi))
+        mtheta = np.ascontiguousarray(-mx * np.cos(ophi) * np.sin(otheta) - my * np.sin(ophi) * np.sin(otheta) + np.cos(otheta) * mz)
+        mr_normalized = np.ascontiguousarray(mr / self.m_maxima)
+        mrminor_normalized = np.ascontiguousarray(mrminor / self.m_maxima)
+        mphi_normalized = np.ascontiguousarray(mphi / self.m_maxima)
+        mtheta_normalized = np.ascontiguousarray(mtheta / self.m_maxima)
+
+        # Save all the data to a vtk file which can be visualized nicely with ParaView
+        data = {"m": (mx, my, mz), "m_normalized": (mx_normalized, my_normalized, mz_normalized), "m_rphiz": (mr, mphi, mz), "m_rphiz_normalized": (mr_normalized, mphi_normalized, mz_normalized), "m_rphitheta": (mrminor, mphi, mtheta), "m_rphitheta_normalized": (mrminor_normalized, mphi_normalized, mtheta_normalized)}
+        pointsToVTK(
+            vtkname, ox, oy, oz, data=data
+        )
+
```

Comment:
With the new JSON framework, you don't have to define `as_dict` and `from_dict` for the `DipoleField` class, and the default methods from `Optimizable `should be sufficient.